### PR TITLE
SMS Survey in the new reminders framework: 1

### PR DESCRIPTION
**Pull request by gcapalbo**
The first step in adding sms survey functionality into the new reminders framework, this just captures all the necessary information in the UI and stores it into the SMSSurveyContent model for the schedule.

@orangejenny

### CODE REVIEW

File: corehq/messaging/scheduling/views.py [hunk 1]
```diff
@@ -143,6 +144,9 @@ def get_scheduled_broadcast(self, broadcast_id):
 
     def get_scheduled_broadcast_activate_ajax_response(self, active_flag, broadcast_id):
         broadcast = self.get_scheduled_broadcast(broadcast_id)
+        if not self.can_use_inbound_sms and broadcast.schedule.memoized_uses_sms_survey:
+            return HttpResponseBadRequest()
```
Comment (orangejenny) [inline corehq/messaging/scheduling/views.py]: Is it worth adding the reason for the failure to the request?

File: corehq/messaging/scheduling/static/scheduling/js/conditional_alert_list.js [hunk 2]
```diff
@@ -70,10 +70,11 @@ hqDefine("scheduling/js/conditional_alert_list", function() {
                     "targets": [4],
                     "render": function(data, type, row) {
                         var id = row[row.length - 1];
+                        var editable = row[row.length - 4];
```
Comment (orangejenny) [inline corehq/messaging/scheduling/static/scheduling/js/conditional_alert_list.js]: The indexing here (and in broadcasts_list.js) is getting hard to follow. What would you think of changing the ajax response to return a dictionary instead of a list?

**Review by orangejenny — APPROVED** [review]
Looks good - a couple comments, but nothing blocking.

File: corehq/messaging/scheduling/views.py (hunk 1 above)
Comment (gcapalbo) [reply corehq/messaging/scheduling/views.py]: we shouldn't actually ever hit this because the ui will restrict it but added the reason in case it ever helps in debugging: https://github.com/dimagi/commcare-hq/pull/19217/commits/343ea2865dd5333ef649efdd0248f6f3af39297f

File: corehq/messaging/scheduling/static/scheduling/js/conditional_alert_list.js (hunk 2 above)
Comment (gcapalbo) [reply corehq/messaging/scheduling/static/scheduling/js/conditional_alert_list.js]: agree, i thought that was a limitation on datatables but i guess it's not: 
https://github.com/dimagi/commcare-hq/pull/19217/commits/ea139f6587f2ce99ee1909af129d91f1011ed5f3
https://github.com/dimagi/commcare-hq/pull/19217/commits/79e070c0aab9d3694759640915a39d352cef76f7

**Review by orangejenny — APPROVED** [review]
